pkg/controller/operators/catalog/operator.go

```diff
@@ -912,41 +917,44 @@ func validateSourceType(logger *logrus.Entry, in *v1alpha1.CatalogSource) (out *
 		err = fmt.Errorf("unknown sourcetype: %s", sourceType)
 	}
 	if err != nil {
+		logger.WithError(err).Error("error validating catalog source type")
 		out.SetError(v1alpha1.CatalogSourceSpecInvalidError, err)
 		return
 	}
 
 	// The sourceType is valid, clear all status (other than status conditions array) if there's existing invalid spec reason
 	if out.Status.Reason == v1alpha1.CatalogSourceSpecInvalidError {
+		logger.Info("catalog source has invalid spec")
 		out.Status = v1alpha1.CatalogSourceStatus{
 			Conditions: out.Status.Conditions,
 		}
 	}
 	continueSync = true
-
 	return
 }
 
 func (o *Operator) syncConfigMap(logger *logrus.Entry, in *v1alpha1.CatalogSource) (out *v1alpha1.CatalogSource, continueSync bool, syncError error) {
+	logger = logger.WithFields(logrus.Fields{
+		"configmap.namespace": in.Namespace,
+		"configmap.name":      in.Spec.ConfigMap,
+	})
+	logger.Info("checking catsrc configmap state")
+
 	out = in
 	if !(in.Spec.SourceType == v1alpha1.SourceTypeInternal || in.Spec.SourceType == v1alpha1.SourceTypeConfigmap) {
 		continueSync = true
+		logger.Info("catalog type is neither internal not configmap - bailing without error")
 		return
 	}
 
 	out = in.DeepCopy()
 
-	logger = logger.WithFields(logrus.Fields{
-		"configmap.namespace": in.Namespace,
-		"configmap.name":      in.Spec.ConfigMap,
-	})
-	logger.Info("checking catsrc configmap state")
-
 	var updateLabel bool
 	// Get the catalog source's config map
 	configMap, err := o.lister.CoreV1().ConfigMapLister().ConfigMaps(in.GetNamespace()).Get(in.Spec.ConfigMap)
 	// Attempt to look up the CM via api call if there is a cache miss
 	if apierrors.IsNotFound(err) {
+		logger.Info("configmap not found in cache - attempting with live client")
```

Comment:
n.b. this is not a safe thing to do, as you no longer operate against a consistent view of cluster state that existed in one point in time (your lister is behind - but you got an event for this configmap? how?)

Reply:
I see what you mean. Thanks for calling that out. I'll poke at it to see if we can just re-enqueue and let the next iteration pick it up from the cache, rather than try to jump ahead of it. Though the code here can get so convoluted its hard to know what taking this jenga piece out will do to the tower XD

Reply:
btw, how's Bill treating you?

Reply:
It's a wild and whacky time! Best of luck with this Jenga :)